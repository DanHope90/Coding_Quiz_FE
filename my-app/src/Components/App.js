/* eslint-disable react/jsx-no-constructed-context-values */
import "../Styles/App.css";
import React, { useState, useEffect } from "react";
import axios from "axios";
import { Route, Routes, BrowserRouter as Router } from "react-router-dom";
import NavBar from "./NavBar";
import Layout from "./Layout";
import Home from "./Home";
import TakeAQuiz from "./TakeAQuiz";
import SavedQuizzes from "./SavedQuizzes";
import Register from "./Register";
import Login from "./Login";
import TestQuiz from "./TestQuiz";
import Context from "../Utils/Context";

function App() {
  // Below sets the initial alert.
  const initialState = {
    alert: { message: "", isSuccess: false },
  };

  const [alert, setAlert] = useState(initialState.alert);

  // Below employs an empty array to store all quizzes, useEffect makes an axios request
  // to populate said array with all quizzes.
  const [availableQuizzes, setAvailableQuizzes] = useState([]);

  useEffect(() => {
    axios
      .get("http://localhost:4000/api/quizzes")
      .then(({ data }) => setAvailableQuizzes(data))
      .catch(() => setAlert({ message: "Axios promise rejected", isSuccess: false }));
  }, []);

  // Below presents a "Loading..." if the database isn't yet
  // accessed. It should only display for a second or two.
  if ((availableQuizzes.length === 0)) {
    return <div>Loading....</div>;
  }

  return (
    <Context.Provider value={{ availableQuizzes }}>
      <div className="app">
        <Router>
          <div className="navbar"><NavBar /></div>
          <Layout>
            <Routes>
              <Route path="/" element={<Home />} />
              <Route path="/take-a-quiz" element={<TakeAQuiz alert={alert} />} />
              <Route path="/saved-quizzes" element={<SavedQuizzes />} />
              <Route path="/register" element={<Register />} />
              <Route path="/login" element={<Login />} />
              <Route path="/test-quiz" element={<TestQuiz />} />
            </Routes>
          </Layout>
        </Router>
      </div>
    </Context.Provider>
  );
}

export default App;
