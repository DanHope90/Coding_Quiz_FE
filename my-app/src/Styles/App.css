body {
    overflow: hidden;
    height: 100vh;
}

main {
    overflow-y: auto;
}

aside {
    flex: 1 0 10%;
}

.navbar {
    display: flex;
    height: 100%
}
