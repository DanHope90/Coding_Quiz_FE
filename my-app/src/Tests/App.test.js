import React from "react";
import renderer from "react-test-renderer";
import App from '../Components/App';

test("renders the app correctly", () => {
  const testRenderer = renderer.create(<App />);
  expect(testRenderer).toMatchSnapshot();
});
