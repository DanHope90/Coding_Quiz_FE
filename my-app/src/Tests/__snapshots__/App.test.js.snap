// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app correctly 1`] = `
<div>
  Loading....
</div>
`;
